Percy cli

andressalinero opened the pull request:
Refactor Percy.cs to use CLI

andressalinero committed:
adding gitignore file

andressalinero committed:
update gitignore

andressalinero committed:
removing Percy Agent references

andressalinero committed:
updated snapshot method, renamed variables to follow Percy SDK naming standard as per Java SDK

andressalinero committed:
added HealtCheck

andressalinero committed:
updating fetchDom

andressalinero committed:
Update FetchDOM & PostSnapshot

andressalinero committed:
Update version

andressalinero committed:
Refactor method name to follow naming convention

ArchitWagle committed:
Corrections to snapshot post, minor bugs.

andressalinero committed:
Merge pull request #1 from ArchitWagle/percy_cli

Corrections to snapshot post, fixed minor bugs.

samirans89 commented on percy-csharp-selenium/Percy.cs:
```diff
@@ -15,63 +15,82 @@ namespace percy_csharp_selenium
+        // Maybe get the CLI server address if not Set the CLI server address 
+        //could be moved to percy-csharp-selenium Environment
+        private String PERCY_SERVER_ADDRESS = System.Environment.GetEnvironmentVariable("PERCY_SERVER_ADDRESS")!=null?
+                System.Environment.GetEnvironmentVariable("PERCY_SERVER_ADDRESS") : "http://localhost:5338";
 
         // Environment information like the programming language, browser, & SDK versions
         private Environment env;
```
We need to have an equivalent of this.
```
 // Determine if we're debug logging
private boolean PERCY_DEBUG = System.getenv("PERCY_LOGLEVEL").equals("debug");
```


```
// for logging 

private String LABEL = "[\u001b[35m" + (PERCY_DEBUG ? "percy:java" : "percy") + "\u001b[39m]";

```

samirans89 commented on percy-csharp-selenium/Percy.cs:
```diff
@@ -15,63 +15,82 @@ namespace percy_csharp_selenium
+
+                return true;
+            } catch (Exception ex) {
+                Console.WriteLine("\nException Caught!");	
```
need equivalent of this.
```
 log("Percy is not running, disabling snapshots");
```

samirans89 reviewed (changes requested):
Yet to go through the complete PR. Here are in the initial changes requested.

samirans89 commented on percy-csharp-selenium/Percy.cs:
```diff
@@ -15,63 +15,82 @@ namespace percy_csharp_selenium
-                using HttpClient client = new HttpClient();
-                var response = await client.GetAsync("http://localhost:5338/" + AGENTJS_FILE);
-
+        private async Task<Boolean> Healthcheck() {
```
can we return simple boolean `true` or `false` from this method
and lets do `Healthcheck().Result;` currently done in constructor inside this method itself.

samirans89 commented on percy-csharp-selenium/Percy.cs:
```diff
@@ -161,6 +175,38 @@ public void Snapshot(String name, List<int> widths, int minHeight, bool enableJa
+
+
+                HttpContent httpEntity = response.Content;
+                String domString = httpEntity.ReadAsStringAsync().Result;
```
.Result is done here as well as in the caller method of `FetchPercyDOM()`.
do we really need to do this twice?

samirans89 commented on percy-csharp-selenium/Percy.cs:
```diff
@@ -161,6 +175,38 @@ public void Snapshot(String name, List<int> widths, int minHeight, bool enableJa
+            } catch (Exception ex) {
+                isPercyEnabled = false;
+                Console.WriteLine("[percy] Something went wrong attempting to fetch DOM: " + ex.Message);
+
```
we need to have this in the MVP. Debugging info is crucial for us when this goes customer facing.

```
 if (PERCY_DEBUG) { log(e.getMessage()); }
```

samirans89 commented on percy-csharp-selenium/Percy.cs:
```diff
@@ -161,6 +175,38 @@ public void Snapshot(String name, List<int> widths, int minHeight, bool enableJa
+        * the DOM. Without it, snapshots cannot be captured.
+        */
+        private async Task<string> FetchPercyDOM() {
+        
```
lets add the condition - same in the Java one:
```
if (!domJs.trim().isEmpty()) { return domJs; }
```

samirans89 commented on percy-csharp-selenium/Percy.cs:
```diff
@@ -172,82 +218,83 @@ public void Snapshot(String name, List<int> widths, int minHeight, bool enableJa
             }
 
-            string base_url = "http://localhost:5338/percy/snapshot";
+            var  param_dict = new Dictionary<string, object>(){
```
query:
does the dictionary instance do the job of using the ContentType = JSON

just wanted to understand which part handles 
```
        StringEntity entity = new StringEntity(json.toString(), ContentType.APPLICATION_JSON);
```

samirans89 commented on percy-csharp-selenium/Percy.cs:
```diff
@@ -172,82 +218,83 @@ public void Snapshot(String name, List<int> widths, int minHeight, bool enableJa
+            };
+
+
+            var res = HttpPostPercySnapshot(param_dict).Result;
```
let's implement equivalent error handling here OR within the HttpPostPercySnapshot.
```
} catch (Exception ex) {            if (PERCY_DEBUG) { log(ex.toString()); }            log("Could not post snapshot " + name);        }

```

samirans89 replied:
also, i understand the same is done in Java but is defining 
```
var res =
```
needed?

andressalinero committed:
check for dom to be fetched

andressalinero committed:
update
Merge branch 'percy_cli' of github-BStack:andressalinero/percy-csharp-selenium into percy_cli
